### PR TITLE
fix: crash on assignment chain

### PR DESCRIPTION
**Pull request by HerrCai0907**
When creating tee assignment, compiler use incorrect this type (type of return value).
It causes crash when return type is not `usize`.
Fixes: #2674

### CODE REVIEW

**Review by MaxGraey — APPROVED** [review]
LGTM
Just fix lint error

File: src/compiler.ts [hunk 1]
```diff
@@ -5811,14 +5811,14 @@ export class Compiler extends DiagnosticEmitter {
...
           let returnTypeRef = returnType.toRef();
-          let tempThis = flow.getTempLocal(returnType);
+          let tempThis = flow.getTempLocal(thisType);
           let ret = module.block(null, [
             this.makeCallDirect(setterInstance, [
-              module.local_tee(tempThis.index, thisExpr, returnType.isManaged),
+              module.local_tee(tempThis.index, thisExpr, /* isManaged=*/false, thisType.toRef()), // thisType is managed but here it must be alive
               valueExpr
             ], valueExpression),
             this.makeCallDirect(getterInstance, [
```
[the remark this answers is not in the document]
Comment (CountBleck) [reply src/compiler.ts:5820]: I do realize you didn't write this, but I wanted to point out what I noticed.

[the remark this answers is not in the document]
Comment (HerrCai0907) [reply src/compiler.ts:5820]: Maybe we can use temp variable to store it?

[the remark this answers is not in the document]
Comment (CountBleck) [reply src/compiler.ts:5820]: Yeah, `valueExpr` should be tee'd to a temp local in the setter call, and the getter call should be replaced with a local.get to that local, *if I'm correct about not calling that getter*.

**Comment by HerrCai0907** [comment]
further issue see #2770

**Pull request merged**